admin/load.php

```diff
@@ -613,12 +639,80 @@ function perflab_deactivate_plugin() {
+					sprintf(
+						'<a href="%s">%s</a>',
+						esc_html( menu_page_url( PERFLAB_MODULES_SCREEN, false ) ),
+						esc_html__( 'Performance Lab Settings', 'performance-lab' )
+					)
+				);
+			echo '</p></div>';
+			return;
+		}
+	}
+
+	if ( ! empty( $standalone_plugins_with_active_modules ) ) {
+		$module_data                 = perflab_get_modules();
+		$standalone_plugins_file_map = perflab_get_standalone_plugins_file_map();
+		$modules                     = array();
+		foreach ( $standalone_plugins_with_active_modules as $standalone_plugins_slug => $module_slug ) {
+			if (
+				isset( $module_data[ $module_slug ] ) &&
+				(
+					! is_plugin_installed( $standalone_plugins_file_map[ $standalone_plugins_slug ] ) ||
+					! is_plugin_active( $standalone_plugins_file_map[ $standalone_plugins_slug ] )
+				)
+			) {
+				$modules[] = esc_attr( $module_data[ $module_slug ]['name'] );
+			}
+		}
+
+		$modules_count = count( $modules );
+		if ( $modules_count < 1 ) {
+			return;
+		} elseif ( 1 === $modules_count ) {
+			echo '<div class="notice notice-warning is-dismissible">';
+				printf(
+					'%s %s',
+					sprintf(
+						/* translators: Performance module */
+						'<p>' . esc_html__( 'Performance Lab has detected usage of "%s" module that have standalone plugin equivalents. Installing and activating a performance plugin will disable the legacy module equivalent. This will not impact functionality of the module.', 'performance-lab' ) . '</p>',
+						esc_attr( $modules[0] )
+					),
+					sprintf(
+						/* translators: Button text */
+						'<p class="perflab-button-wrapper"><button type="button" class="button button-primary perflab-install-active-plugin">%s</button><span class="dashicons dashicons-update hidden"></span></p>',
+						esc_html__( 'Migrate legacy modules to standalone plugins', 'performance-lab' )
+					)
+				);
+			echo '</div>';
+		} else {
+			$module_name = '<ol>';
+			foreach ( $modules as $module ) {
+				$module_name .= "<li>$module</li>";
+			}
+			$module_name .= '</ol>';
+			echo '<div class="notice notice-warning is-dismissible">';
+				printf(
+					'%s %s %s %s',
+					'<p>' . esc_html__( 'Performance Lab has detected usage of modules that have standalone plugin equivalents. Please see the available plugins below. Installing and activating a performance plugin will disable the legacy module equivalent. This will not impact functionality of the module.', 'performance-lab' ) . '</p>',
+					'<strong>' . esc_html__( 'Available standalone plugins:', 'performance-lab' ) . '</strong>',
+					wp_kses_post( $module_name ),
+					sprintf(
+						'<p class="perflab-button-wrapper"><button type="button" class="button button-primary perflab-install-active-plugin">%s</button><span class="dashicons dashicons-update hidden"></span></p>',
+						esc_html__( 'Migrate legacy modules to standalone plugins', 'performance-lab' )
+					)
+				);
+			echo '</div>';
```

Comment:
A few recommendations here:
* Can you combine the two conditions so that the common output is shared between them? We should only need if-else for the parts that differ.
* It would be great if you could directly print HTML where possible rather than putting it in PHP strings and echoing them. Also, the `sprintf()` calls with all the HTML are difficult to follow, we can probably avoid them.
    * It's easier to read.
    * It's more error-proof as we'll have HTML syntax highlighting.